packages/zapp/console/src/components/flytegraph/ReactFlow/customNodeComponents.tsx

```diff
@@ -260,11 +344,45 @@ export const ReactFlowCustomTaskNode = ({ data }: any) => {
     }
   };
 
+  const renderMapTasks = (externalResourcesByPhase) => {
+    return (
+      <div style={mapTaskWrapper}>
+        {RENDER_ORDER.map((phase, id) => {
+          if (!externalResourcesByPhase.has(phase)) {
+            return null;
+          }
+          let color = nodePhaseColorMapping[phase].color;
+          if (selectedMapTask && selectedMapTask !== phase) {
+            color = 'gray';
+          }
+
+          const key = `${id}-${phase}`;
+          return (
+            <MapTaskItem
+              numberOfTasks={externalResourcesByPhase.get(phase).length}
+              color={color}
+              phase={phase}
+              setSelectedMapTask={setSelectedMapTask}
+              setSelectedNode={setSelectedNode}
+              key={key}
+            />
+          );
+        })}
+      </div>
+    );
+  };
+
   return (
-    <div onClick={handleClick}>
-      {data.taskType ? renderTaskType() : null}
+    <div onClick={handleNodeClick}>
+      {data.nodeExternalResourcesByPhase
```

Comment:
? Do we plan to show task name inside the node rectangle in future ?

Reply:
I don't think it's gonna live _inside_ the rectangle in near future

Reply:
May be let's sync on it with Jason and Honxgun on Monday, it's definitely not a P1. but I'm not sure how I feel about "replacing" type with array task name